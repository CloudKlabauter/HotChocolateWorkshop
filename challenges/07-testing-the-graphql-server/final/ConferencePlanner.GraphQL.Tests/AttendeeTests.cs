using System;
using System.Threading.Tasks;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.DependencyInjection;
using ConferencePlanner.GraphQL;
using ConferencePlanner.GraphQL.Attendees;
using ConferencePlanner.GraphQL.Data;
using ConferencePlanner.GraphQL.Sessions;
using ConferencePlanner.GraphQL.Speakers;
using ConferencePlanner.GraphQL.Tracks;
using ConferencePlanner.GraphQL.Types;
using HotChocolate;
using HotChocolate.Execution;
using Snapshooter.Xunit;
using Xunit;

namespace ConferencePlanner.GraphQL.Tests;

public class AttendeeTests
{
    [Fact]
    public async Task Attendee_Schema_Changed()
    {
        // arrange
        // act
        ISchema schema = await new ServiceCollection()
            .AddPooledDbContextFactory<ApplicationDbContext>(
                options => options.UseInMemoryDatabase("Data Source=conferences.db"))
            .AddGraphQL()
            .AddQueryType(d => d.Name("Query"))
                .AddTypeExtension<AttendeeQueries>()
            .AddMutationType(d => d.Name("Mutation"))
                .AddTypeExtension<AttendeeMutations>()
            .AddType<AttendeeType>()
            .AddType<SessionType>()
            .AddType<SpeakerType>()
            .AddType<TrackType>()
            .AddGlobalObjectIdentification()
            .BuildSchemaAsync();

        // assert
        schema.Print().MatchSnapshot();
    }

    [Fact]
    public async Task RegisterAttendee()
    {
        // arrange
        IRequestExecutor executor = await new ServiceCollection()
            .AddPooledDbContextFactory<ApplicationDbContext>(
                options => options.UseInMemoryDatabase("Data Source=conferences.db"))
            .AddGraphQL()
            .AddQueryType(d => d.Name("Query"))
                .AddTypeExtension<AttendeeQueries>()
            .AddMutationType(d => d.Name("Mutation"))
                .AddTypeExtension<AttendeeMutations>()
            .AddType<AttendeeType>()
            .AddType<SessionType>()
            .AddType<SpeakerType>()
            .AddType<TrackType>()
            .AddGlobalObjectIdentification()
            .BuildRequestExecutorAsync();

        // act
        IExecutionResult result = await executor.ExecuteAsync(@"
           mutation RegisterAttendee {
               registerAttendee(
                   input: {
                       emailAddress: ""michael@chillicream.com""
                           firstName: ""michael""
                           lastName: ""staib""
                           userName: ""michael3""
                       })
               {
                   attendee {
                       id
                   }
               }
           }");

        // assert
        result.ToJson().MatchSnapshot();
    }
}