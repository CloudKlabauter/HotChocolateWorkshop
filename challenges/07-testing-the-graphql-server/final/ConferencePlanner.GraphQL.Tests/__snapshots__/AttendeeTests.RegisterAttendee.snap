{
  "data": {
    "registerAttendee": {
      "attendee": {
        "id": "QXR0ZW5kZWUKaTE="
      }
    }
  }
}
